Source/DafnyCore/Resolver.cs

```diff
@@ -9744,6 +9744,9 @@ public CheckDividedConstructorInit_Visitor(Resolver resolver)
           dependencies.AddVertex(idt);
           foreach (Formal p in ctor.Formals) {
             AddDatatypeDependencyEdge(idt, p.Type, dependencies);
+            foreach (Type t in p.Type.TypeArgs) {
+              AddDatatypeDependencyEdge(idt, t, dependencies);
+            }
```

Comment:
Is this needed only for datatypes?  (not for codatatypes?)

Reply:
This is a good question, and I do not know. Let me summarize:

1. With respect to equality support specifically, it is indeed not needed because codatatypes do not support equaity, so there's no need to dig deeper.
2. The dependency graph computation has a comment: "_The dependencies of interest among codatatypes are just the top-level types of parameters_" 

So I think that it is best not to change the computation for the codatatypes and remember that the name of dependency graph is slightly misleading because not all dependencies are included.